packages/tools/fluid-runner/src/exportFile.ts

```diff
@@ -0,0 +1,115 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import * as fs from "fs";
+import path from "path";
+import { ITelemetryLogger } from "@fluidframework/common-definitions";
+import { Loader } from "@fluidframework/container-loader";
+import { createLocalOdspDocumentServiceFactory } from "@fluidframework/odsp-driver";
+import { ChildLogger, PerformanceEvent } from "@fluidframework/telemetry-utils";
+// eslint-disable-next-line import/no-internal-modules
+import { FileLogger } from "./logger/FileLogger";
+import { getArgsValidationError } from "./getArgsValidationError";
+import { IFluidFileConverter, isCodeLoaderBundle } from "./codeLoaderBundle";
+import { FakeUrlResolver } from "./fakeUrlResolver";
+
+export type IExportFileResponse = IExportFileResponseSuccess | IExportFileResponseFailure;
+
+interface IExportFileResponseSuccess {
+    success: true;
+}
+
+interface IExportFileResponseFailure {
+    success: false;
+    eventName: string;
+    errorMessage: string;
+}
+
+const clientArgsValidationError = "Client_ArgsValidationError";
+
+export async function exportFile(
+    codeLoader: string,
+    inputFile: string,
+    outputFolder: string,
+    scenario: string,
+    telemetryFile: string,
+): Promise<IExportFileResponse> {
+    if (fs.existsSync(telemetryFile)) {
+        return {
+            success: false,
+            eventName: clientArgsValidationError,
+            errorMessage: `Telemetry file already exists [${telemetryFile}]`,
+        };
+    }
+
+    const fileLogger = new FileLogger(telemetryFile, 50);
+
+    const logger = ChildLogger.create(fileLogger, "LocalSnapshotRunnerApp",
+        { all: { Event_Time: () => Date.now() } });
+
+    try {
+        return await PerformanceEvent.timedExecAsync(logger, { eventName: "ExportFile" }, async () => {
+            // eslint-disable-next-line @typescript-eslint/no-require-imports, @typescript-eslint/no-var-requires
+            const codeLoaderBundle = require(codeLoader);
+            if (!isCodeLoaderBundle(codeLoaderBundle)) {
+                const eventName = clientArgsValidationError;
+                const message = "Code loader bundle is not of type CodeLoaderBundle";
+                logger.sendErrorEvent({
```

Comment:
I'd think the caller would log. Otherwise why add eventName to IExportFileResponse?